kong/cmd/utils/prefix_handler.lua

```diff
@@ -667,6 +677,10 @@ local function prepare_prefix(kong_config, nginx_custom_template_path, skip_writ
       read_err = tostring(read_err or "unknown error")
       return nil, "failed reading custom nginx template file: " .. read_err
     end
+
+    if nginx_custom_template_path == "spec/fixtures/custom_nginx.template" then
+      template_env_opts.test_custom = true
```

Comment:
This design using a global variable makes the behavior more non-transparent (and makes an assumption that we only enter the function once, which would be a surprise for unit tests). We should rather make `template_env_opts` a parameter.

Reply:
agree. I've made the template rendering environment options a parameter and initialized them based on the input param "test_mode" of prepare_prefix. This is set directly by start.lua and depends on the `test_mode` argument which is passed by helpers/start_kong.

I think it's better to still allow passing the template file path to start Kong with, just in case we need a different one in the future for some reason.

Reply:
Maybe we have to modify the cmd. I prefer to use string tags instead of magic numbers.